### PR TITLE
fix: If allow_origins_by_regex is not nil, should be considered to allow_origins_by_regex only

### PR DESCRIPTION
**Pull request by Qiuqiu0505**
### Description

Fixes https://github.com/apache/apisix/issues/9027
 -- If allow_origins_by_regex is not nil, should be considered to allow_origins_by_regex only.

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Review by qihaiyan — APPROVED** [review]

**Comment by qihaiyan** [comment]
Is this PR ready to merge? @spacewander

File: apisix/plugins/cors.lua [hunk 1]
```diff
@@ -297,13 +297,13 @@ end
 
 function _M.header_filter(conf, ctx)
     local req_origin =  ctx.original_request_origin
-    -- Try allow_origins first, if mismatched, try allow_origins_by_regex.
+    -- If allow_origins_by_regex is not nil, should be matched to it only
     local allow_origins
-    allow_origins = process_with_allow_origins(conf.allow_origins, ctx, req_origin)
-    if not match_origins(req_origin, allow_origins) then
-        allow_origins = process_with_allow_origins_by_regex(conf, ctx, req_origin)
+    allow_origins = process_with_allow_origins_by_regex(conf, ctx, req_origin)
```
Comment (spacewander) [inline apisix/plugins/cors.lua:302]: Look like we can skip this function call by checking the `conf.allow_origins_by_regex == nil` first. For example,

```
if conf.allow_origins_by_regex ~= nil then
```

Comment (Qiuqiu0505) [reply apisix/plugins/cors.lua:302]: image

This function named  process_with_allow_origins_by_regex has checked  whether allow_origins_by_regex is  nil. 
Does it need to be checked again before the method invocation?

Comment (spacewander) [reply apisix/plugins/cors.lua:302]: Yes. We can move it out, and avoid the repeated `and conf.allow_origins_by_regex == nil` check below.

Comment (Qiuqiu0505) [reply apisix/plugins/cors.lua]: Thanks  and I has changed it.

**Comment by spacewander** [comment]
Let's merge master to make CI pass.

**Comment by spacewander** [comment]
https://github.com/apache/apisix/actions/runs/4446285829/jobs/7839374950?pr=9028
```
#   Failed test 't/plugin/cors2.t TEST 5: regex specified match - header Vary ok'
#   at /apisix/test-nginx/lib/Test/Nginx/Socket.pm line 1011.
#          got: 'Via'
#     expected: 'Via, Origin'
# Looks like you failed 1 test of 35.
[02:10:05] t/plugin/cors2.t ..
```

Please make the CI pass, thanks!

**Comment by Qiuqiu0505** [comment]
> https://github.com/apache/apisix/actions/runs/4446285829/jobs/7839374950?pr=9028
> 
> ```
> #   Failed test 't/plugin/cors2.t TEST 5: regex specified match - header Vary ok'
> #   at /apisix/test-nginx/lib/Test/Nginx/Socket.pm line 1011.
> #          got: 'Via'
> #     expected: 'Via, Origin'
> # Looks like you failed 1 test of 35.
> [02:10:05] t/plugin/cors2.t ..
> ```
> 
> Please make the CI pass, thanks!

**Comment by Qiuqiu0505** [comment]
@spacewander  I have just updated it. Thanks

**Review by spacewander — APPROVED** [review]

**Review by qihaiyan — APPROVED** [review]

**Review by spacewander — APPROVED** [review]

**Comment by qihaiyan** [comment]
Merging needs another approving of reviewer with write access. @spacewander

**Review by soulbird — APPROVED** [review]

**Review by leslie-tsang — APPROVED** [review]

**Comment by KID-G** [comment]
Hi,
Thank you for your contribution to Apache APISIX!
Can you tell me your email address? We have prepared electronic certificates for new contributors.

**Pull request merged**